src/libs/actions/Policy.js

```diff
@@ -406,6 +408,73 @@ function update(policyID, values, shouldGrowl = false) {
         });
 }
 
+/**
+ * @param {String} policyID The id of the workspace / policy
+ * @param {String} name The new workspace name
+ */
+function updateWorkspaceName(policyID, name) {
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    name: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                },
+
+                // Clear errorFields.name in case the user didn't dismiss the error
+                errorFields: {
+                    name: null,
+                },
+                name,
+            },
+        },
+    ];
+    const successData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    name: null,
+                },
+                errorFields: {
+                    name: null,
+                },
+            },
+        },
+    ];
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    name: null,
+                },
+                errorFields: {
+                    name: {
+                        [DateUtils.getMicroseconds()]: 'Sorry, there was an unexpected problem updating your workspace name.',
```

Comment:
I know this is not ready yet, but this is a string the user can see, so it needs to be localized.

Reply:
Updated. I'll get a copy check too.